src/Microsoft.VisualStudio.ProjectSystem.Managed/ImmutableCollectionLinqExtensions.cs

```diff
@@ -41,6 +40,7 @@ internal static class ImmutableCollectionLinqExtensions
             return false;
         }
 
+        [return: MaybeNull]
         public static T FirstOrDefault<T, TArg>(this ImmutableArray<T> immutableArray, Func<T, TArg, bool> predicate, TArg arg)
```

Comment:
This is the recommended approach for indicating that an output with unconstrained generic type can be null.